contrib/src/databases.rs

```diff
@@ -0,0 +1,84 @@
+pub extern crate r2d2;
+
+use std::collections::BTreeMap;
+use std::error::Error;
+use std::io;
+use std::marker::{Send, Sized};
+use rocket::config::{self, ConfigError, Value};
+
+use self::r2d2::ManageConnection;
+
+#[cfg(feature = "diesel_sqlite")]
+pub extern crate diesel;
+
+#[cfg(feature = "pg_pool")]
+pub extern crate postgres;
+#[cfg(feature = "pg_pool")]
+pub extern crate r2d2_postgres;
+
+pub struct DatabaseConfig<'a> {
+    pub url: &'a str,
+    pub pool_size: u32,
+    pub extras: Option<&'a BTreeMap<String, Value>>,
+}
+
+pub fn database_config<'a>(name: &str, from: &'a config::Config) -> Option<DatabaseConfig<'a>> {
```

Comment:
Don't forget to document any `pub` items! Even during review, it's useful to have a tiny blurb.

Reply:
Good call, thanks!